src/tree_schema_helpers.c

```diff
@@ -1479,30 +1479,24 @@ lysc_node_children(const struct lysc_node *node, uint16_t flags)
 API const struct lysc_node *
 lysc_node_children_full(const struct lysc_node *node, uint16_t flags)
 {
-    switch (node->nodetype) {
-    case LYS_CONTAINER:
-        return ((struct lysc_node_container *)node)->child;
-    case LYS_CHOICE:
-        return (struct lysc_node *)((struct lysc_node_choice *)node)->cases;
-    case LYS_CASE:
-        return ((struct lysc_node_case *)node)->child;
-    case LYS_LIST:
-        return ((struct lysc_node_list *)node)->child;
-    case LYS_RPC:
-    case LYS_ACTION:
+    if (!node) {
+        return NULL;
+    }
+
+    if (node->nodetype == LYS_RPC || node->nodetype == LYS_ACTION) {
         if (flags & LYS_CONFIG_R) {
             return (struct lysc_node *)&((struct lysc_action *)node)->output;
         } else {
             /* LYS_CONFIG_W, but also the default case */
             return (struct lysc_node *)&((struct lysc_action *)node)->input;
         }
-    case LYS_INPUT:
-    case LYS_OUTPUT:
-        return ((struct lysc_action_inout *)node)->data;
-    case LYS_NOTIF:
-        return ((struct lysc_notif *)node)->data;
-    default:
-        return NULL;
+    } else {
+        struct lysc_node **children = lysc_node_children_p(node, flags);
+        if (children) {
+            return *children;
+        } else {
+            return NULL;
+        }
```

Comment:
Why not simply call `lysc_node_children()` instead of this code?

Reply:
you are right, changed